core/chaincode/handler.go

```diff
@@ -743,8 +743,9 @@ func (h *Handler) HandleGetStateByRange(msg *pb.ChaincodeMessage, txContext *Tra
 				startKey = metadata.Bookmark
 			}
 		}
-		rangeIter, err = txContext.TXSimulator.GetStateRangeScanIteratorWithMetadata(namespaceID,
-			startKey, getStateByRange.EndKey, paginationInfo)
+		pageSize := paginationInfo["limit"].(int32)
```

Comment:
No need to create a map and then lookup for a key...simply extract `metadata.PageSize` inside the above if loop?

Reply:
Nice suggestion. Removed some 35 LOC.